### ISSUE TITLE
Validité des seuils de taux plein / taux réduit pour le calcul des prélèvements sociaux revenus de remplacement

### ISSUE THREAD

**Issue by pzuldp** [openfisca/openfisca-france#2028]
Hello hello !

Je suis le fan numéro un d'OpenFisca, mais je viens de rencontrer un problème.

### Qu'ai-je fait ?

J'ai examiné la validité des formules de calcul des prélèvements sociaux sur les revenus de remplacements (ainsi que leur inversion dans openfisca-france-data)

### À quoi m'attendais-je ?

A ce que les seuils pour les taux réduits et taux plein soit une fonction du nombre de parts fiscales du foyer fiscal, *divisé par deux*.

### Que s'est-il passé en réalité ?

Les seuils pour les taux réduits et taux plein soit une fonction du nombre de parts fiscales du foyer fiscal (non divisé par deux).

### Contexte

Je m'identifie plus en tant que :

- [x] Contributeur·e : je contribue à OpenFisca France.
- [ ] Développeur·e : je crée des outils qui utilisent OpenFisca France.
- [x] Économiste : je réalise des simulations avec des données.
- [ ] Mainteneur·e : j'intègre les contributions à OpenFisca France.
- [ ] Autre : _(ajoutez une description du contexte dans lequel vous utilisez OpenFisca)_.

**Comment by pzuldp** [comment]
Pour être plus précis : ces deux lignes apparaissent assez souvent dans les formules de CSG/CRDS : 
```
        seuil_exoneration = seuils.seuil_rfr1.seuil_rfr1 + (nbptr - 1) * seuils.seuil_rfr1.demi_part_suppl_rfr1
        seuil_reduction = seuils.seuil_rfr2.seuil_rfr2 + (nbptr - 1) * seuils.seuil_rfr2.demi_part_suppl_rfr2
```
mais il me semble qu'il faudrait à chaque fois remplacer par 
```
        seuil_exoneration = seuils.seuil_rfr1.seuil_rfr1 + (nbptr - 1)/2 * seuils.seuil_rfr1.demi_part_suppl_rfr1
        seuil_reduction = seuils.seuil_rfr2.seuil_rfr2 + (nbptr - 1)/2 * seuils.seuil_rfr2.demi_part_suppl_rfr2
```
si j'en juge par [cette source](https://www.demarches.interieur.gouv.fr/particuliers/csg-crds-revenus-activite-remplacement) par exemple ?

**Comment by pzuldp** [comment]
je vais tagger quelques personnes, on ne sait jamais.
@bfabre01 @SophieIPP @NolwennLoisel @lukas-puschnig @clallemand

**Comment by pzuldp** [comment]
si quelqu'un.e se sent d'attaque pour une review, c'est par ici : 
https://github.com/openfisca/openfisca-france/pull/2030

**Comment by pzuldp** [comment]
en fait une série de tests existait déjà, mais était incapable de détecter l'erreur car tous les cas-types n'avaient qu'une seule part fiscale (en fait, pas de part fiscale spécifiée, d'où le défaut de 1)

**Comment by clallemand** [comment]
Hello, j'aterris ici au cours de la review de ta PR associée. Désolée j'arrive un peu tard mais je ne comprends pas où tu trouves le "x2"... Il y a sûrement un truc que je ne comprends pas, même dans la source si dessus

**Comment by SophieIPP** [comment]
Je me pose la même question que @clallemand . De ce que je comprends, on compare le RFR du foyer, pondéré par la composition du foyer, aux différents seuils (taux nul, taux réduits).

**Comment by sylvainipp** [comment]
Il me semble que le *2 est simplement du au fait que le montant correspond à des demi-parts, alors que le terme précédent (nbptr - 1) est en nombre de part. (J'ai aussi l'impression que c'est sur le seuil que porte la pondération par taille du foyer et pas sur le RFR, avec des majorations par demi-part qui sont constantes au moins dans la fameuse source de Paul).

**Comment by pzuldp** [comment]
Je suis désolé si je me suis trompé... mais je crois que c'est juste
Comme l'indique @sylvainipp , il me semble simplement que 
NBPTR = nombre de parts ; d'où (NBPTR - 1) = "nombre de parts supplémentaires"
alors que l'incrément `seuils.seuil_rfr1.demi_part_suppl_rfr1` fait référence à un supplément par DEMI-part supplémentaire.
Pour un foyer à deux parts, il faut donc faire :
`seuil_rfr + demi_part_suppl  + demi_part_suppl`, soit `seuil_rfr + (nbptr - 1) * 2 * demi_part_suppl`

**Comment by pzuldp** [comment]
@SophieIPP je suis d'accord mais c'est la pondération qui pose problème. @clallemand dans le tableau de référence cité, on a bien, pour un foyer à deux parts, le seuil "de base" auquel on ajoute 2 fois l'incrément par demi parts.

**Comment by clallemand** [comment]
Ah yes c'est bon je comprends, c'est ça ! Merci j'étais trop concentré à faire x2 dans le barème et je ne comprenais pas... C'est très clair et très juste je pense !
